app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt

```diff
@@ -376,6 +367,7 @@ class MainActivity : AppCompatActivity() {
             VisionSafetyManager.destroy()
             VisionManager.stop()
             VisionManager.destroy()
+            vision_view.onPause()
```

Comment:
@kiryldz as far as i understood from the API description `onPause`/`onResume` calls are required from developer (though it's really confusing and stackoverflow has a lot of question if it's _really_ mandatory to call them). Anyway, seems we might also need to put some recommendations about that to the docs, if we think the calls are necessary

Reply:
Calling `onPause` (https://developer.android.com/reference/android/opengl/GLSurfaceView#onPause()) for sure stops render thread and optionally could preserve GL context (it is not that important for us). After `onPause()` method `surfaceDestroyed(SurfaceHolder holder)` is called. In my understanding at this point we free all memory. 
Calling `onResume`
(https://developer.android.com/reference/android/opengl/GLSurfaceView#onResume()) will re-create GL context (in case it was detached) and allocate all drawers, textures etc again.
So I see nothing confusing in logic and totally agree that we should put those recommendations to docs and leave it up to developers. Avoiding `onPause` and `onResume` will not cause any disasters but potentially could cause GPU leak.